src/coreclr/jit/importer.cpp

```diff
@@ -3825,7 +3825,7 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
             // on a local are probably pretty useless anyway, so we probably don't care.
 
             op1 = gtNewOperNode(interlockedOperator, genActualType(callType), op1, op2);
-            op1->gtFlags |= GTF_GLOB_REF | GTF_ASG;
+            op1->gtFlags |= GTF_GLOB_REF | GTF_ASG | GTF_EXCEPT;
```

Comment:
I think we should we only add `GTF_EXCEPT` if `fgAddrCouldBeNull(op1)` is `false`.

Reply:
You are right, I will do add this. 
However, It does not matter nowadays because we ignore  `GTF_EXCEPT` value after the importation phase. Morph resets it and sets the correct value using `OperMayThrow`, it happens here: https://github.com/dotnet/runtime/blob/88be24b566d6408417c27a20dc1df91516d01a59/src/coreclr/jit/morph.cpp#L13248-L13259

Reply:
Sure, but eventually we'd like to have correct flags after the importation phase and enforce them with the checker. We already have some optimizations between importation and morph and if we decide to add more, we may be forced to fix the flags.